### PR TITLE
feat(cli): tkn-result would port-forward to service/tekton-results-api-service automatically

### PR DESCRIPTION
**Pull request by xinnjie**
# Changes
If Results API server address not specified, tkn-result would port-forward to `service/tekton-results-api-service` automatically.
It helps a lot when `service/tekton-results-api-service` is not exported.

Test case:
*  Results API server address specified, no functionality is changed.
    * If address is invalid,  cli will timeout.
    * If address is valid,  cli will query successfully.
*  Results API server address not  specified, cli queries Results API service successfully without `Kubectl port-forward` manually.

/kind feature

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you review them:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] [Tested your changes locally](https://github.com/tektoncd/results/blob/main/docs/DEVELOPMENT.md) (if this is a code change)
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including functionality, content, code)
- [x] Has a kind label. You can add a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [x] Release notes block below has been updated with any user-facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contain the string "action required" if the change requires additional action from users switching to the new release

# Release Notes
```release-note
[CLI]No need to port-forward `service/tekton-results-api-service` manually. If Results API server address not specified and not disable explicitly, tkn-result will port-forward to service/tekton-results-api-service automatically.
```

### CODE REVIEW

**Review by khrm — COMMENTED** [review]
We can probably add this. There are a lot of CLI prs remaining. We need to merge these.

**Comment by xinnjie** [comment]
> We can probably add this. There are a lot of CLI prs remaining. We need to merge these.

Could you give the pr a lgtm if it is ok to add the feature? Thanks a lot. @khrm

**Comment by sayan-biswas** [comment]
@xinnjie I'll check this out today.

**Review by khrm — COMMENTED** [review]
We probably need this behind a flag not automatically port-forward. @sayan-biswas @avinal What do you think?

**Comment by sayan-biswas** [comment]
> We probably need this behind a flag not automatically port-forward. @sayan-biswas @avinal What do you think?

Agreed!! 

Although there's a pre-condition to this automatic behaviour (server address not configured), but still I feel a flag would provide more explicit control to the user.

**Comment by avinal** [comment]
> We probably need this behind a flag not automatically port-forward. @sayan-biswas @avinal What do you think?

I have similar point as Sayan, but I am more in favor of automatic since a condition is already defined. That said, a flag would be useful from a user's control perspective. 

@xinnjie The changes look good to me, but I am unable to test due to a problem with my setup. I will put my test results if I fix my setup.

**Review by avinal — COMMENTED** [review]
Finally, I was able to test it using codespaces. Changes are working as expected. 
<details>
<summary>Test results</summary>

### Config I used.

```yaml
address:
service_account:
  name: tekton-results-debug
  namespace: tekton-pipelines
ssl:
  roots_file_path: /tmp/results.crt
  server_name_override: tekton-results-api-service.tekton-pipelines.svc.cluster.local
```
</details>

**Comment by xinnjie** [comment]
@avinal @sayan-biswas @khrm 
Add a flag  `-disable-port-forwarding` to disable port-forwarding while `address` is empty. That said, when `address` is empty, port-forwarding is the default behavior, users can choose to disable it explicitly.

I prefer port-forward automatically (when `address`  is empty), because there is a use case that `tekton-results-api-service` is not exported outside the cluster,  and people who do not export the `tekton-results-api-service`  need auto port-forwarding more.

**Comment by xinnjie** [comment]
`portforward` flag added.


Test case:
* `portforward` not set in config file and commandline, `addr` not set. Port forward run as expected. 
* When `portforward: false` is added to config file.
```
> ./tkn-results  list -                   
Dial: context deadline exceeded
Error: context deadline exceeded
Usage:
  tkn-results list [flags] <parent>
...
```

* When `portforward` not set in config file and  `--portforward=false` set in command line:
```
> ./tkn-results --portforward=false  list -                   
Dial: context deadline exceeded
Error: context deadline exceeded
Usage:
  tkn-results list [flags] <parent>

  <parent>: Parent name to query. This is typically corresponds to a namespace, but may vary depending on the API Server. "-" may be used to query all parents.

```

**Review by avinal — APPROVED** [review]
Tested, working as expected

**Comment by xinnjie** [comment]
Please any one review this PR. It has been 2 weeks now.

**Comment by avinal** [comment]
cc @enarha @sayan-biswas as discussed in the WG call, I have tested it, please put a final review, it is ready to be merged.

**Comment by enarha** [comment]
@xinnjie Apologies for the delay on this one. Can you please rebase and fix the conflicts (I want to avoid doing that manually) and I'll make sure we address the PR in the next couple of days. Thank you again.

**Comment by xinnjie** [comment]
Rebased.

**Review by enarha — COMMENTED** [review]
Apart from the vendoring issue, the rest is LGTM.

File: tools/tkn-results/internal/portforward/portforward.go
Comment (avinal) [reply tools/tkn-results/internal/portforward/portforward.go:11]: There is a PR #550 is open for bumping the version. And issue for vendoring #551. This PR is good to go.

Comment (xinnjie) [reply tools/tkn-results/internal/portforward/portforward.go:11]: And sorry for not updating `go.mod` in this PR.

**Pull request merged**
